src/bindings/python/src/pyopenvino/core/common.cpp

```diff
@@ -88,6 +90,109 @@ ov::Tensor tensor_from_numpy(py::array& array, bool shared_memory) {
     return tensor;
 }
 
+ov::PartialShape partial_shape_from_list(const py::list& shape) {
+    using value_type = ov::Dimension::value_type;
+    ov::PartialShape pshape;
+    for (py::handle dim : shape) {
+        if (py::isinstance<py::int_>(dim)) {
+            pshape.insert(pshape.end(), ov::Dimension(dim.cast<value_type>()));
+        } else if (py::isinstance<py::str>(dim)) {
+            pshape.insert(pshape.end(), Common::dimension_from_str(dim.cast<std::string>()));
+        } else if (py::isinstance<ov::Dimension>(dim)) {
+            pshape.insert(pshape.end(), dim.cast<ov::Dimension>());
+        } else if (py::isinstance<py::list>(dim) || py::isinstance<py::tuple>(dim)) {
+            py::list bounded_dim = dim.cast<py::list>();
+            if (bounded_dim.size() != 2) {
+                throw py::type_error("Two elements are expected in tuple(lower, upper) for dynamic dimension, but " +
+                                     std::to_string(bounded_dim.size()) + " were given.");
+            }
+            if (!(py::isinstance<py::int_>(bounded_dim[0]) && py::isinstance<py::int_>(bounded_dim[1]))) {
+                std::string wrong_type = py::isinstance<py::int_>(bounded_dim[0]) ? bounded_dim[1].get_type().str()
+                                                                                  : bounded_dim[0].get_type().str();
+                throw py::type_error("Incorrect type " + wrong_type + " for dynamic dimension, int is expected.");
+            }
```

Comment:
```suggestion
            if (!(py::isinstance<py::int_>(bounded_dim[0]) && py::isinstance<py::int_>(bounded_dim[1])) {
                std::string wrong_type = bounded_dim[0].get_type().str() + ", " + bounded_dim[1].get_type().str();
                throw py::type_error("Incorrect pair of types (" + wrong_type + ") for dynamic dimension, ints are expected.");
            }
```
Something like this would give better information on which argument min/max (or both) is wrong and why.

Reply:
done